src/custom/pages/Profile/LockedGnoVesting/hooks.ts

```diff
@@ -0,0 +1,129 @@
+import { useCallback, useEffect, useMemo, useState } from 'react'
+import { ContractTransaction } from '@ethersproject/contracts'
+import { CurrencyAmount, Token } from '@uniswap/sdk-core'
+import { useActiveWeb3React } from 'hooks/web3'
+import MERKLE_DROP_ABI from 'abis/MerkleDrop.json'
+import TOKEN_DISTRO_ABI from 'abis/TokenDistro.json'
+import { MerkleDrop, TokenDistro } from 'abis/types'
+import { useSingleCallResult } from 'lib/hooks/multicall'
+import { useTransactionAdder } from 'state/enhancedTransactions/hooks'
+import { useContract } from 'hooks/useContract'
+import { COW as COW_TOKENS } from 'constants/tokens'
+import { SupportedChainId } from 'constants/chains'
+import { OperationType } from 'components/TransactionConfirmationModal'
+import { fetchClaim } from './claimData'
+
+// We just generally use the mainnet version. We don't read from the contract anyways so the address doesn't matter
+const COW = COW_TOKENS[SupportedChainId.MAINNET]
+
+export const MERKLE_DROP_CONTRACT_ADDRESSES: Record<number, string> = {
+  [SupportedChainId.MAINNET]: '0x64646f112FfD6F1B7533359CFaAF7998F23C8c40',
+  [SupportedChainId.RINKEBY]: '0x5444c4AFb2ec7f7367C10F7732b8558650c5899F',
+  [SupportedChainId.XDAI]: '0x3d610e917130f9D036e85A030596807f57e11093',
```

Comment:
This is the wrong address, must be fixed to:

```suggestion
  [SupportedChainId.XDAI]: '0x48D8566887F8c7d99757CE29c2cD39962bfd9547',
```